### PR TITLE
Create CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by nareddyt**
Added a CONTRIBUTING.md as discussed in #174. 

I mostly followed [this project's CONTRIBUTING.md](https://github.com/pulumi/pulumi/blob/master/CONTRIBUTING.md), but also added some FAQs for developing locally. We can migrate those FAQs to the wiki if needed.

It would be great if someone could check my bash `go` commands. I used an IDE (Goland) to setup pomerium in my development environment, so I'm not entirely sure if those commands are correct.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+### Prerequisites
+
+- [Fork the repository](https://help.github.com/en/articles/fork-a-repo)
+- Install [Go 1.12](https://golang.org/doc/install) or later
+- (Optional) Node.js and npm for modifying the docs
+
+### Setup Sources Locally
+
+Download the forked sources to your go workspace with `go get`:
```
[the remark this answers is not in the document]
Comment (desimone) [reply CONTRIBUTING.md]: Just tested running just `make` on a fresh repo with a clean go-mod cache. I think we can just say do `make`. Thoughts?

**Comment by nareddyt** [comment]
Hey, sorry for the delay. I started a new job this week, so i'll be a bit busy. Probably won't be able to address these comments until Sunday... Feel free to take over this issue if you really need the CONTRIBUTING.md done sooner.

I did read your comments and I agree with most of them 👍

**Comment by desimone** [comment]
@nareddyt Totally understand!

I was roughly hoping to have #174 as a whole done by the end of this release (~end of month) so there's no rush. Oh, and a big congrats on the new job!!! 🍾

**Comment by desimone** [comment]
@nareddyt I'm going to close as I think almost all the point have been addressed in other issues. Let me know if that's not the case. Thanks!

**Pull request closed without merging**